flowey/flowey_lib_common/src/junit_publish_test_results.rs

```diff
@@ -25,6 +27,8 @@ flowey_request! {
             junit_xml: ReadVar<Option<PathBuf>>,
             /// Brief string used when publishing the test.
             test_label: String,
+            /// Additional attachments for platforms without JUnit integration (not used on ADO)
```

Comment:
I would prefer it if we kept this junit node small and focused, handling only junit related stuff. All this new functionality should be in its own separate node.

If you'd like, we can then have a meta helper node called `publish_test_results` which encapsulates certain bits of reusable logic wrt. what to do on which backend.

I'd like to keep these abstraction layers clean

Reply:
I thought this would make sense since on ADO, the "junit" node uploads all of these files. So I'm not really changing the scope of this node, I'm just making all of the branches to the same thing. And since there isn't really a logical way to split the ADO junit upload into different layers, I think it makes sense to keep this in the same file. However, maybe we should rename it be just publish_test_results?

Reply:
Right, that's the opposite approach we can take. Changing the charter of this node to be a generic `publish_test_results` node, where callers can give it XML files, directories, whatever... and it'll do the "right thing" on each platform

Reply:
I guess we could rename junit_publish_test_results to ado_junit_publish_test_results and make it ADO-only if you want to split that out, and the rest of the logic (including uploading junit files on github and local) can be in the publish_test_results file

Reply:
Ok I refactored this to split out the ado and github tasks